### ISSUE TITLE
Distributed FFmpeg binary is not GPL compliant

### ISSUE THREAD

**Issue by Brassn** [openaudible/openaudible#384]
The FFmpeg binary currently included in OpenAudible 2.1.1 is compiled with "--enable-nonfree" and using the fkd-aac library which is incompatible to the GPL. The resulting binary cannot be distributed legally.

Also the license file states that FFmpeg was created as a LGPL binary which is (no longer) the case.
Please change your build parameters accordingly.

For reference: https://trac.ffmpeg.org/wiki/Encode/AAC

**Comment by openaudible** [comment]
As far as I know, we're not using any of the non-free libraries. We are not actually encoding or decoding M4A audio, and we don't use the libfdk_aac tag. 

I'll look into building static binaries with the non-free flag off.

**Comment by openaudible** [comment]
I built a new version of ffmpeg with enabe-gpl and disabled-nonfree. (Plus disabled almost everything else.. making in 2/3 smaller.) Works great. 

Need to build for all 3 targets and test.. and will include in the next release. 

```
ffmpeg version N-98034-gb6d7c4c Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.12) 20160609
  configuration: --prefix=/root/ffmpeg_build --pkg-config-flags=--static --extra-libs=-static --extra-cflags=--static --extra-cflags=-I/root/ffmpeg_build/include --extra-ldflags=-L/root/ffmpeg_build/lib --extra-libs='-lpthread -lm' --bindir=/root/bin --enable-gpl --enable-libfreetype --enable-libmp3lame

```
Appreciate the heads up. I'll review the rest of the ffmpeg license docs as well. 
I'll see if I can publish my build scripts.. currently building for amd64 using:
https://github.com/srwareham/docker-ffmpeg-compiler

**Comment by kategray** [comment]
Thank you for filing the ticket, @Brassn.  I was just about to file one myself.

The distributed version of ffmpeg also contains x264.  Distribution of x264 requires either compliance with the GPL, or a commercial license.

`--pkg-config=pkg-config --pkg-config-flags=--static --extra-version=ffmpeg-windows-build-helpers --enable-version3 --disable-debug --disable-w32threads --arch=x86_64 --target-os=mingw32 --cross-prefix=/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32- --enable-libcaca --enable-gray --enable-libtesseract --enable-fontconfig --enable-gmp --enable-gnutls --enable-libass --enable-libbluray --enable-libbs2b --enable-libflite --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopus --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libvorbis --enable-libwebp --enable-libzimg --enable-libzvbi --enable-libmysofa --enable-libopenjpeg --enable-libopenh264 --enable-liblensfun --enable-libvmaf --enable-libsrt --enable-demuxer=dash --enable-libxml2 --enable-libdav1d --enable-libsvthevc --enable-libaom --enable-libvpx --enable-nvenc --enable-nvdec --extra-libs=-lharfbuzz --extra-libs=-lm --extra-libs=-lpthread --extra-cflags=-DLIBTWOLAME_STATIC --extra-cflags=-DMODPLUG_STATIC --extra-cflags=-DCACA_STATIC --enable-amf --enable-libmfx --enable-gpl --enable-avisynth --enable-frei0r --enable-filter=frei0r --enable-librubberband --enable-libvidstab --enable-libx264 --enable-libx265 --enable-libxvid --enable-libxavs --enable-avresample --extra-cflags='-mtune=generic' --extra-cflags=-O3 --enable-static --disable-shared --prefix=/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 --enable-nonfree --enable-decklink --enable-libfdk-aac`

Commercial licenses for x264 are available, but you're probably not interested in paying the cost for something you aren't using.

Please, either release your source code or quit distributing a product built around a GPL version of ffmpeg.  Given the rather heavy use you make of ffmpeg, there's a case to be made that it's a derivative work.

Thank you.

**Comment by openaudible** [comment]
Yeah, I got some help building an LGPL minimal install for all three platforms. Not easy for a java developer! (mvn package didn't work :) ) Will add to next release. I also made a donation to the ffmpeg group (Software in the Public Interest, Inc.) and appreciate their hard work.

**Comment by openaudible** [comment]
Fixed in 2.2

**Comment by bghira** [comment]
you could just make it all GPL and keep the OpenAudible open-source, as, it is in the name (Open) and you're effectively stealing from all these other projects and violating their licenses. and these are only the ones we know about.
